streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -182,10 +152,54 @@ public String toString(final String indent) {
         return sb.toString();
     }
 
+    protected Map<TopicPartition, Long> recordCollectorOffsets() {
+        return Collections.emptyMap();
+    }
+
+    protected void updateOffsetLimits() {
+        log.trace("{} Updating store offset limits", logPrefix);
+        for (final TopicPartition partition : partitions) {
+            try {
+                final OffsetAndMetadata metadata = consumer.committed(partition); // TODO: batch API?
+                stateMgr.putOffsetLimit(partition, metadata != null ? metadata.offset() : 0L);
+            } catch (final AuthorizationException e) {
+                throw new ProcessorStateException(String.format("task [%s] AuthorizationException when initializing offsets for %s", id, partition), e);
+            } catch (final WakeupException e) {
+                throw e;
+            } catch (final KafkaException e) {
+                throw new ProcessorStateException(String.format("task [%s] Failed to initialize offsets for %s", id, partition), e);
+            }
+        }
+    }
+
     /**
      * Flush all state stores owned by this task
      */
     void flushState() {
+        log.trace("{} Flushing state store", logPrefix);
```

Comment:
Well, it's that the same "thing" the manager is just an implementation details?
The idea is, to log that `task.flushState()` was called -- this log is independent of "how" the flushing is done. Does this make sense?

And I thinks it better to keep it within the if -> this los is not about a method got called, but what does actually happen -- and thus only log if an actual flush is done.